src/service/invoice-service.ts

```diff
@@ -420,21 +446,46 @@ export default class InvoiceService {
    * Returns the latest invoice sent to a User that is not deleted.
    * @param toId
    */
-  static async getLatestValidInvoice(toId: number): Promise<BaseInvoiceResponse> {
-    const invoices = (await this.getInvoices({ toId })).records;
+  static async getLatestValidInvoice(toId: number): Promise<Invoice> {
+    const invoices = (await this.getInvoices({ toId }));
     // Filter the deleted invoices
     const validInvoices = invoices.filter(
-      (invoice) => invoice.currentState.state !== InvoiceState[InvoiceState.DELETED],
+      (invoice) =>  InvoiceService.isState(invoice, InvoiceState.DELETED),
     );
     return validInvoices[validInvoices.length - 1];
   }
 
+  /**
+   * TODO
+   * @param userId
+   */
+  public static async getDefaultInvoiceParams(userId: number): Promise<InvoiceUserDefaults> {
```

Comment:
This function has not enough coverage, especially the "normal" flow is missing.

Reply:
fixed